packages/node/src/integrations/http.ts

```diff
@@ -103,7 +104,7 @@ export const instrumentHttp = Object.assign(
       },
 
       ignoreIncomingRequestHook: request => {
-        const url = getRequestUrl(request);
+        const url = request.url;
```

Comment:
The really interesting part here is that TS doesn't throw a type error here. In `ignoreIncomingRequestHook`, the `request` parameter is of type `IncomingMessage` but `getRequestUrl` required the `RequestOptions` type. These two types don't inherit from each other. So not really sure why TS doesn't complain. Maybe because all properties in `RequestOptions` are optional 🤔

Reply:
curious if you feel like checking - if you change the type sig of the getRequestUrl function to anything else does typescript complain?
If so, I think your theory about all-optional may be correct. Good theory.

Reply:
just checking to be sure, is the `url` here the full URL, including protocol? so e.g. `http://...`? As we want these to be full URLs I guess always, if possible.

Reply:
Yup, if I change the type for the `getRequestUrl` input to something else, e.g.

```ts
export function getRequestUrl(requestOptions: { foo?: string }): string
```

I get type errors in both cases.

Reply:
@mydea `url` is NOT the full url, it is instead just the path (and maybe the query params?).

the `node:IncomingMessage` type doesn't have a `protocol`, `hostname`, or `port` attr.
* For protocol you'd have to do something like access `req.socket.encrypted` and infer true === https
* For hostname you'd have to parse the request headers for a Host header, falling back to req.socket.address() if it doesn't exist.
* For port you'd have to access `req.socket.address().port`, I think.

So all that to say, sadly it's difficult to reconstruct the full url. Thoughts?

Reply:
Here's all the available attrs:
<img width="512" alt="Screen Shot 2024-07-16 at 5 40 06 AM" src="https://github.com/user-attachments/assets/56eb8948-445d-4ca6-a61e-4146a730b76a">

Reply:
Yeah, I feared that. IMHO this is OK, but we should maybe reflect this in the jsdoc/variables used for `ignoreIncomingRequests`, e.g. make it `ignoreIncomingRequests(urlPath: string)` or something like this?